### PR TITLE
Generate storage info for pallet grandpa

### PR DESCRIPTION
**Pull request by georgesdib**
Fixes part of #8629

Refactor Grandpa pallet to use `WeakBoundedVec` instead of `Vec`, and generate storage info for that pallet.
Adding `MaxAuthorities` type to bound the `WeakBoundedVec`.

Polkadot companion: paritytech/polkadot#3892

@thiolliere @KiChjang @shawntabrizi

### CODE REVIEW

File: frame/grandpa/src/lib.rs [hunk 1]
```diff
@@ -355,15 +365,25 @@ pub trait WeightInfo {
...
 
+/// Bounded version of `AuthorityList`, `Limit` being the bound
+pub type BoundedAuthorityList<Limit> = WeakBoundedVec<(AuthorityId, AuthorityWeight), Limit>;
+
 /// A stored pending change.
-#[derive(Encode, Decode, TypeInfo)]
-pub struct StoredPendingChange<N> {
+/// `Limit` is the bound for `next_authorities`
+#[derive(Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[codec(mel_bound(Limit: Get<u32>))]
```
Comment (thiolliere) [inline frame/grandpa/src/lib.rs:374]: sidenote: I am surprised it is needed, I though skipping the params didn't remove the normal bounds below.

**Review by thiolliere — APPROVED** [review]

**Review by KiChjang — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Comment by thiolliere** [comment]
I think companion needs to be fixed, test-runtime needs the new configuration

**Comment by georgesdib** [comment]
> I think companion needs to be fixed, test-runtime needs the new configuration

This is done now

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**